File: Models/NhatKySuDung.cs
using System;
using System.Collections.Generic;

namespace WebAPI_NRW.Models.Database;

public partial class NhatKySuDung
{
    public int Id { get; set; }

    public int? TenNguoiDung { get; set; }

    public string HanhDong { get; set; } = null!;

    public string? TinhNang { get; set; }

    public string? DuLieu { get; set; }

    public string? GhiChu { get; set; }

    public DateTime NgayTao { get; set; }

    public DateTime? NgayCapNhat { get; set; }

    public int NguoiTao { get; set; }

    public int? NguoiCapNhat { get; set; }

    public virtual NguoiDung? NguoiCapNhatNavigation { get; set; }

    public virtual NguoiDung NguoiTaoNavigation { get; set; } = null!;

    public virtual NguoiDung? TenNguoiDungNavigation { get; set; }
}

File: src/types/nguoi-dung/nhat-ky-su-dung.ts
/* ========== REQUEST (CLIENT --> API) ========== */
// Thêm dữ liệu
export interface AddNhatKySuDungRequest {
  tenNguoiDung?: number;   // số ID người dùng
  hanhDong: string;
  tinhNang?: string;
  duLieu?: string;
  ghiChu?: string;
  ngayTao: string;
  nguoiTao: string;        // BE nhận string
}

// Cập nhật dữ liệu
export interface UpdateNhatKySuDungRequest {
  tenNguoiDung?: number;
  hanhDong: string;
  tinhNang?: string;
  duLieu?: string;
  ghiChu?: string;
  ngayCapNhat?: string;
  nguoiCapNhat?: string;
}

/* ========== RESPONSE (API --> CLIENT) ========== */
export interface NhatKySuDungResponse {
  id: number;
  tenNguoiDung: string;
  hanhDong: string;
  tinhNang: string;
  duLieu: string;
  ghiChu?: string;
  ngayTao: string;
  ngayCapNhat?: string;
  nguoiTao: string;
  nguoiCapNhat?: string;
}
